### ISSUE TITLE
[BUG] division by zero error when calculating F1 score

### ISSUE THREAD

**Issue by pygabr** [alan-turing-institute/sktime#10768]
Hi, again on the ClaSPTransformer. I noted a "division by zero" error when using the F1 score as scoring metric.
It appears to be caused by the use of precision and recall in the calculation, instead of the confusion matrix components (from _clasp_numba.py):

```
pr = tp / (tp + fp)
re = tp / (tp + fn)

f1 = 2 * (pr * re) / (pr + re)

```
when tp = 0, it all breaks down generating the error. Using the confusion matrix (as sklearn does) it would lead to f1 = 0.

**To Reproduce**
```
import numpy
from sktime.transformations.clasp import ClaSPTransformer

a=numpy.arange(100)
c=ClaSPTransformer(scoring_metric = 'F1')
c.fit_transform(a)
```

Thank you

**Comment by aminemanai2003** [comment]
﻿Hi, I reproduced this on current `main` with NumPy 2.3.5 and Numba 0.62.1. The precision/recall form of the F1 calculation reaches `0 / 0` when there are no true positives.

I have a local fix using `2 * TP / (2 * TP + FP + FN)`, returning zero when the denominator is zero. I also added a regression test through `ClaSPTransformer(scoring_metric="F1")`. The new test and the three existing ClaSP tests pass.

I used AI assistance for the implementation and testing. Would this scope be suitable for a PR?
